core/src/main/scala/org/apache/spark/metrics/MetricsConfig.scala

```diff
@@ -97,5 +90,23 @@ private[spark] class MetricsConfig(val configFile: Option[String]) extends Loggi
       case None => propertyCategories.getOrElse(DEFAULT_PREFIX, new Properties)
     }
   }
-}
 
+  private[this] def loadPropertiesFromFile(): Unit = {
+    var is: InputStream = null
+    try {
+      is = configFile match {
+        case Some(f) => new FileInputStream(f)
+        case None => Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_METRICS_CONF_FILENAME)
+      }
+
+      if (is != null) {
+        properties.load(is)
+      }
+    } catch {
+      case e: Exception => logError("Error loading configure file", e)
```

Comment:
can you also include what the file we're trying to load is? (`spark.metrics.conf` or the default conf file?)